modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/actions/data/sql/client/AbstractSQLAction.java

```diff
@@ -365,14 +365,21 @@ private CallableStatement getPreparedCall(Connection conn, SQLDatasource datasou
     private ArrayList<ColumnDefinition> getColumnDefinitions(ResultSet rs)
             throws SQLException {
         ArrayList<ColumnDefinition> columnDefs = new ArrayList<>();
+        ArrayList<String> columnNames = new ArrayList<>();
```

Comment:
A Set would be the more suitable data structure for this. Specially when there's a large number of columns, it's O(1) lookup would be useful for the "contains" operation, compared to ArrayList's O(n) contains operation.

Reply:
Fixed with a56e89440acb2461848b604b4d25ef85f0862843. Thanks